controllers/sync.go

```diff
@@ -602,21 +604,13 @@ func (r *YtsaurusReconciler) Sync(ctx context.Context, resource *ytv1.Ytsaurus)
 
 		logger.Info("Ytsaurus update was finished and Ytsaurus is running now")
 		err := ytsaurus.SaveClusterState(ctx, ytv1.ClusterStateRunning)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		err = ytsaurus.SaveObserverGeneration(ctx, resource.ObjectMeta.Generation)
-		return ctrl.Result{}, err
+		return ctrl.Result{Requeue: true}, err
```

Comment:
Why do we want to requeue here and in `ClusterStateReconfiguration`? If I understand correctly after change of status.  I suppose you just explicitly  set the requeue here.

Reply:
Spec might already be updated yet another time.
It's easier to requeue than inline all handling at place.

At next iteration operator does final check that we have reached desired state and do final update of observed generation

or starts next update and also does update of observed generation